x-pack/legacy/plugins/lens/public/editor_frame_plugin/editor_frame/editor_frame.test.tsx

```diff
@@ -807,30 +807,37 @@ describe('editor_frame', () => {
       await waitForPromises();
 
       expect(mockDatasource.getPublicAPI).toHaveBeenCalledWith(
-        datasource1State,
-        expect.anything(),
-        'first'
+        expect.objectContaining({
+          state: datasource1State,
+          setState: expect.anything(),
+          layerId: 'first',
+        })
       );
       expect(mockDatasource2.getPublicAPI).toHaveBeenCalledWith(
-        datasource2State,
-        expect.anything(),
-        'second'
+        expect.objectContaining({
+          state: datasource2State,
+          setState: expect.anything(),
+          layerId: 'second',
+        })
       );
       expect(mockDatasource2.getPublicAPI).toHaveBeenCalledWith(
-        datasource2State,
-        expect.anything(),
-        'third'
+        expect.objectContaining({
+          state: datasource2State,
+          setState: expect.anything(),
+          layerId: 'third',
+        })
       );
     });
 
     it('should give access to the datasource state in the datasource factory function', async () => {
       const datasourceState = {};
+      const dateRange = { fromDate: 'now-1w', toDate: 'now' };
       mockDatasource.initialize.mockResolvedValue(datasourceState);
       mockDatasource.getLayers.mockReturnValue(['first']);
 
       mount(
         <EditorFrame
-          {...getDefaultProps()}
+          {...{ ...getDefaultProps(), dateRange }}
```

Comment:
Isn't this the same as `{...getDefaultProps()} dateRange={dateRange}`?

Reply:
Yup. I'll change it.